src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -672,10 +672,16 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
      *
      * @param ContainerInterface $container The service container
      *
+     * @throws \DomainException If the container is not of type ContainerBuilder
+     *
      * @return DelegatingLoader The loader
      */
     protected function getContainerLoader(ContainerInterface $container)
     {
+        if (!$container instanceof ContainerBuilder) {
+            throw new \DomainException(sprintf('Container must be an instance of Symfony\Component\DependencyInjection\ContainerBuilder, got %s', get_class($container)));
```

Comment:
Im lost on symfony and exceptions. Based on this [ref](https://github.com/symfony/symfony/blob/f5f1e103c4573bf539f9069d1458b75821f0be30/src/Symfony/Bundle/FrameworkBundle/Templating/Loader/TemplateLocator.php#L69) i think we should do that.